packages/block-editor/src/hooks/style.js

```diff
@@ -202,6 +243,45 @@ export const withBlockControls = createHigherOrderComponent(
 	'withToolbarControls'
 );
 
+/**
+ * Override the default block element to include duotone styles.
+ *
+ * @param  {Function} BlockListBlock Original component
+ * @return {Function}                Wrapped component
+ */
+const withElementsStyles = createHigherOrderComponent(
+	( BlockListBlock ) => ( props ) => {
+		const elements = props.attributes.style?.elements;
+		if ( ! elements ) {
+			return <BlockListBlock { ...props } />;
+		}
+		const blockElementsContainerIdentifier = `wp-block-elements-container-${ useInstanceId(
+			BlockListBlock
+		) }`;
+		const styles = compileElementsStyles(
```

Comment:
The style generated for the link color needs to include `!important`, same logic as in [the server](https://github.com/WordPress/gutenberg/pull/31524/files#r634252402).

Test case:

- Use the TT1 theme.
- Add a link color to a paragraph.

The expected result is that it works, but it doesn't because the theme styles have higher specificity `.editor-styles-wrapper .wp-block a { color: var(--wp--style--color--link, var(--global--color-primary)); }`.